lib/pbench/server/filetree.py

```diff
@@ -161,23 +146,28 @@ def create(tarball: Path, controller: "Controller") -> "Tarball":
         proper place along with the md5 companion file. It returns the new
         Tarball object.
         """
-        destination = controller.path / tarball.name
+        # Make sure that the tarball name ends with ".tar.xz" and complain
+        # if not.
+        name = Tarball.stem(tarball)
+        if name != tarball.name[:-7]:
+            raise BadFilename(tarball)
```

Comment:
I see, now.  But, I think the test would pass with a `.sixltr` suffix, as well; so, the code doesn't quite do what the comment says.  Perhaps,
```
        name = Tarball.stem(tarball)
        if name + ".tar.xz" != tarball.name:
            raise BadFilename(tarball)
```
But I don't like sprinkling the `".tar.xz"` throughout the code any more than we have to (i.e., that's kinda the _point_ of `Tarball.stem()`...).

Perhaps it would be good to have `Tarball.stem()` do the validation instead of doing it here?